internal/alertmanager/upstream.go

```diff
@@ -15,7 +15,7 @@ var (
 )
 
 // NewAlertmanager creates a new Alertmanager instance
-func NewAlertmanager(name, uri string, timeout time.Duration) error {
+func NewAlertmanager(name, uri string, timeout time.Duration, proxyRequests bool) error {
```

Comment:
I'd suggest using the [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) pattern instead. While reviewing consumer code, it's not clear what the lonesome `false` means.

Reply:
An example of what this might look like: https://gist.github.com/terinjokes/18c666ec4a52cd5bd10df29b4e6f95a6